### PR TITLE
Geolocation API: http://www.w3.org/TR/geolocation-API/

### PR DESCRIPTION
**Pull request by slegrand45**
This is a proposal for a binding to Geolocation API. Regarding the code:

- May be the `geolocation` object should be added to the current `navigator` object attributes?
- The last two parameters (for the error function and for the options) of `getCurrentPosition` and `watchPosition` should be optional. I don't know how to specify this?

### CODE REVIEW

File: lib/geolocation.mli [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      match code with
+      | Geolocation.TIMEOUT -> Firebug.console##debug(msg)
+      | _ -> ()
+    in
+    geo##getCurrentPosition(f_success, f_error, options) *)
+
+type positionErrorCode =
+    PERMISSION_DENIED
+  | POSITION_UNAVAILABLE
+  | TIMEOUT
```
Comment (hhugo) [inline lib/geolocation.mli]: This doesn't seems correct. The spec says.
```
  [NoInterfaceObject]
  interface PositionError {
    const unsigned short PERMISSION_DENIED = 1;
    const unsigned short POSITION_UNAVAILABLE = 2;
    const unsigned short TIMEOUT = 3;
    readonly attribute unsigned short code;
    readonly attribute DOMString message;
  };
```

File: lib/geolocation.mli [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  method heading : float Js.opt Js.readonly_prop
+  method speed : float Js.opt Js.readonly_prop
+end
+
+class type position = object
+  method coords : coordinates Js.t Js.readonly_prop
+  method timestamp : Js.date Js.readonly_prop
+end
+
+class type positionOptions = object
```
Comment (hhugo) [inline lib/geolocation.mli]: optional options should either be `Js.writeonly_prop` or `Js.optdef Js.readonly_prop`

File: lib/geolocation.ml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    -> (positionError Js.t -> unit)
+    -> positionOptions Js.t
+    -> unit Js.meth
+  method watchPosition : (position Js.t -> unit)
+    -> (positionError Js.t -> unit)
+    -> positionOptions Js.t
+    -> int Js.meth
+  method clearWatch : int -> unit Js.meth
+end
+
```
Comment (hhugo) [inline lib/geolocation.ml]: please introduce an abstract type for `watchId` and use it for `clearWatch` and `watchPosition`

File: lib/geolocation.ml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+class type positionError = object
+  method code : positionErrorCode Js.readonly_prop
+  method message : Js.js_string Js.t Js.readonly_prop
+end
+
+class type geolocation = object
+  method getCurrentPosition : (position Js.t -> unit)
+    -> (positionError Js.t -> unit)
```
Comment (hhugo) [inline lib/geolocation.ml]: You should use Js.callback : `(position Js.t -> unit) Js.callback`

**Comment by slegrand45** [comment]
Should be better now

File: lib/geolocation.ml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+end
+
+class type positionOptions = object
+  method enableHighAccuracy : bool Js.writeonly_prop
+  method timeout : int Js.writeonly_prop
+  method maximumAge : int Js.writeonly_prop
+end
+
+class type positionError = object
+  method _PERMISSION_DENIED : positionErrorCode Js.readonly_prop
```
Comment (hhugo) [inline lib/geolocation.ml]: should be `_PERMISSION_DENIED_`.
The syntax extension remove the first `_` if it exists and everything after the last `_`

**Comment by slegrand45** [comment]
Underscore added

**Comment by hhugo** [comment]
thank you

**Pull request merged**